backend/src/v2/component/launcher_v2.go

```diff
@@ -550,14 +550,14 @@ func fetchNonDefaultBuckets(
 		// a. If imported, artifact bucket can still be specified in kfp-launcher config (not implemented)
 		// b. If imported, artifact bucket can not be in kfp-launcher config, in this case, return no session and rely on env for aws config
 		if !strings.HasPrefix(artifact.Uri, defaultBucketConfig.PrefixedBucket()) {
-			nonDefaultBucketConfig, err := objectstore.ParseBucketConfigForArtifactURI(artifact.Uri)
-			if err != nil {
-				return nonDefaultBuckets, fmt.Errorf("failed to parse bucketConfig for output artifact %q with uri %q: %w", name, artifact.GetUri(), err)
+			nonDefaultBucketConfig, parseErr := objectstore.ParseBucketConfigForArtifactURI(artifact.Uri)
+			if parseErr != nil {
+				return nonDefaultBuckets, fmt.Errorf("failed to parse bucketConfig for output artifact %q with uri %q: %w", name, artifact.GetUri(), parseErr)
 			}
 			// If the run is cached, it will be in the same bucket but under a different path, re-use the default bucket
 			// session in this case.
 			if (nonDefaultBucketConfig.Scheme == defaultBucketConfig.Scheme) && (nonDefaultBucketConfig.BucketName == defaultBucketConfig.BucketName) {
-				nonDefaultBucketConfig.Session = defaultBucketConfig.Session
```

Comment:
can we also use upstream's version of fetchNonDefaultBuckets(() ? 
https://github.com/kubeflow/pipelines/blob/2.2.0/backend/src/v2/component/launcher_v2.go#L528-L565

It's mostly comment changes, and the `err` var has been reworded.